openvasd/openvasd.c

```diff
@@ -704,15 +702,22 @@ openvasd_result_new (unsigned long id, gchar *type, gchar *ip_address,
   result->ip_address = g_strdup (ip_address);
   result->hostname = g_strdup (hostname);
   result->oid = g_strdup (oid);
-  result->port = port;
-  result->protocol = g_strdup (protocol);
   result->message = g_strdup (message);
   result->detail_name = g_strdup (detail_name);
   result->detail_value = g_strdup (detail_value);
   result->detail_source_name = g_strdup (detail_source_name);
   result->detail_source_type = g_strdup (detail_source_type);
   result->detail_source_description = g_strdup (detail_source_description);
 
+  if (!g_strcmp0 (type, "host_detail"))
+    result->port = g_strdup ("general/Host_Details");
+  else if (port == NULL || (!g_strcmp0 (port, "0") && protocol))
+    result->port = g_strdup_printf ("general/%s", protocol);
+  else if (port && protocol)
```

Comment:
The `port` check is redundant, right? It's compared to NULL above.

Hard to review this kind of patch where the formatting changes are mixed in.

Reply:
Yes, sorry for the formatting. I have run clang-format thinking on my changes, but a lot of code was changed. Next time I will check/fix the format after committing.